lib/kennel/syncer.rb

```diff
@@ -244,16 +234,19 @@ def print_diff(diff)
           new = Utils.pretty_inspect(new)
         end
 
-        if use_diff
-          out.puts "  #{type}#{field}"
-          out.puts(diff(old, new).map { |l| "    #{l}" })
-        elsif (old + new).size > 100
-          out.puts "  #{type}#{field}"
-          out.puts "    #{old} ->"
-          out.puts "    #{new}"
-        else
-          out.puts "  #{type}#{field} #{old} -> #{new}"
-        end
+        message =
+          if use_diff
+            "  #{type}#{field}\n" +
+              diff(old, new).map { |l| "    #{l}" }.join("\n")
+          elsif (old + new).size > 100
+            "  #{type}#{field}\n" \
+              "    #{old} ->\n" \
+              "    #{new}"
+          else
+            "  #{type}#{field} #{old} -> #{new}"
+          end
+
+        Kennel.out.puts truncate_diff(message)
```

Comment:
I like this diff – apart from the fact that I would prefer truncate_diff to be in a different file.

Reply:
utils was meant for these 1-off methods that don't belong anywhere
once they get to complex or there are multiple that could be a class I try to move them out

Reply:
Suuuuure....  not sure why you mention that. It feels like you've answered a question that I didn't ask. Maybe I'm just tired.

Reply:
ah you meant truncate_diff not the utils method ...
the code was in syncer before so 🤷 
also having a new file for 5 loc makes reading the code harder
... as long as methods are self-contained and easy to read I don't mind having a lot of them, it just gets ugly if they all interact with each-other and share state

Reply:
Some people would call things that are self contained and don't share state with other things, a "class" :-p